app/main/autoUpdate.ts

```diff
@@ -0,0 +1,143 @@
+
+interface RealUpdateInfo extends UpdateInfo {
+    downloadedFile: string;
+}
+
+async function getRemoteContent(url: string, binary = false) {
+    const options: AxiosRequestConfig = {};
+
+    if (binary) {
+        options.responseType = 'arraybuffer';
+    }
+
+    const { data } = await axios.get(url, options);
+    return data;
+}
+
+function getSha256Sum(path: string): Promise<string> {
+    const sum = createHash('sha256');
+    const stream = fs.createReadStream(path);
+
+    stream.on('data', (data) => sum.update(data));
+
+    return new Promise((resolve, reject) => {
+        stream.on('end', () => resolve(sum.digest('hex')));
+        stream.on('error', reject);
+    });
+}
+
+function getVerifier(path: string): Promise<Verify> {
+    const verifier = createVerify('RSA-SHA256');
+    const stream = fs.createReadStream(path);
+
+    stream.on('data', (data) => verifier.update(data));
+
+    return new Promise((resolve, reject) => {
+        stream.on('end', () => resolve(verifier));
+        stream.on('error', reject);
+    });
+}
+
+const releasesFeed = `https://github.com/${build.publish.owner}/${build.publish.repo}/releases/download`;
+
+function getVerificationFunctions(
+    { downloadedFile: filePath, releaseName, path: fileName }: RealUpdateInfo // UpdateInfo interface doesn't seem to be aligned with actual content.
+) {
+    const releaseFileUrl = `${releasesFeed}/${releaseName}/${fileName}`;
+
+    return {
+        async verifyChecksum() {
+            const remoteHash = (
+                await getRemoteContent(`${releaseFileUrl}.hash`)
+            ).trim();
+            const localHash = await getSha256Sum(filePath);
+
+            if (localHash !== remoteHash) {
+                throw new Error(
+                    `Local checksum of update does not match remote checksum (local: ${localHash}, remote: ${remoteHash}).`
+                );
+            }
+        },
+        async verifySignature() {
+            const [remoteSig, pubKey] = await Promise.all([
+                getRemoteContent(`${releaseFileUrl}.sig`, true),
+                getRemoteContent(publicKeyUrl),
+            ]);
+
+            const verifier = await getVerifier(filePath);
+            const success = verifier.verify(
+                Buffer.from(pubKey, 'ascii'),
+                remoteSig
+            );
+
+            if (!success) {
+                throw new Error('Signature verification failed.');
+            }
+        },
+    };
+}
+
+const handleUpdateDownloaded = (mainWindow: BrowserWindow) => async (
+    info: UpdateInfo
+) => {
+    mainWindow.webContents.send(updateDownloaded);
+
+    const { verifyChecksum, verifySignature } = getVerificationFunctions(
+        info as RealUpdateInfo
+    );
+
+    try {
+        await Promise.all([verifyChecksum(), verifySignature()]);
+        mainWindow.webContents.send(updateVerified);
+    } catch (e) {
+        log.error('Could not update application due to:', e);
```

Comment:
Does this way of logging align with the logging branch that is open elsewhere? The logs from the auto updater should also end up in a log file to be useful to us.

Reply:
On this branch `electron-log` is used, but I agree that we should align. I talked this through with @shjortConcordium, and he suggested that he'd adapt in his branch if it's merged later. Otherwise if we get logging merged before, I'll include it in this PR.